### PR TITLE
Feature: AFX shorthand for @if and @process / generate keys

### PR DESCRIPTION
**Pull request by mhsdesign**
resolves: https://github.com/neos/fusion-afx/issues/29

Automatically generates key for `@if` and `@process` when no one is specified.
Multiple meta properties transpile to: '@if.if_1' and '@if.if_2'

**Demo**
```html
# afx
<Vendor.Site:Prototype @if={eel1} image.@if={eel2} />

```

```ts
# fusion
Vendor.Site:Prototype {
    @if.if_1 = ${eel1}
    image.@if.if_1 = ${eel2}
}
```

**What I did**
before the attributes are handed into the big loop and converted to fusion, some keys are modified.
Each attribute key string that ends with either `@if` or `@process` will get an auto generated key.

**Improvements i did not do yet.**
I think the function `generatePathForShorthandFusionMetaPath` could be integrated in the for loop succeeding it (we would save one loop then) but i did not want to create spaghetti yet. (The generation relies on a `global` index list wich will decide the next number - easier would be a random number or a not path aware incrementing count. (a simple i++))

btw watch out:
The global editorconfig from neos removes wanted whitespaces from the AfxServiceTest.php...

**How to verify it**
added some tests ; )

**Checklist**

- [x] Tests have been created, run and adjusted as needed

Resolves: https://github.com/neos/fusion-afx/issues/29

### CODE REVIEW

File: Neos.Fusion.Afx/Classes/Service/AfxService.php [hunk 1]
```diff
@@ -127,6 +128,32 @@ protected static function propListToFusion($payload, $attributePrefix, $indentat
...
+     * @return array
+     */
+    protected static function generatePathForShorthandFusionMetaPath(array $attributes): array
+    {
+        // holds the incrementing index per attribute path
+        $indexes = [];
+        foreach ($attributes as &$attribute) {
+            if ($attribute['type'] !== 'prop')
+                continue;
+
```
Comment (jonnitto) [inline Neos.Fusion.Afx/Classes/Service/AfxService.php:142]:
```suggestion
            if ($attribute['type'] !== 'prop') {
                continue;
            }

```

File: Neos.Fusion.Afx/Tests/Functional/AfxServiceTest.php [hunk 2]
```diff
@@ -423,11 +475,11 @@ public function complexChildrenAreRenderedAsArrayIgnoringWhitespaceInBetween():
...
 <h1>
-    
+
     <strong>foo</strong>
-    
+
     <i>bar</i>
-        
+
 </h1>
```
Comment (jonnitto) [inline Neos.Fusion.Afx/Tests/Functional/AfxServiceTest.php:483]: I think this need to be reverted

File: Neos.Fusion.Afx/Tests/Functional/AfxServiceTest.php [hunk 3]
```diff
@@ -824,11 +876,11 @@ public function multilineCommentsAreIgnored(): void
...
 <h1>
-<!-- 
-    comment 
+<!--
+    comment
     with
-    multiple 
-    lines 
+    multiple
+    lines
```
Comment (jonnitto) [inline Neos.Fusion.Afx/Tests/Functional/AfxServiceTest.php:883]: I think this need to be reverted

**Review by jonnitto — CHANGES_REQUESTED** [review]

**Comment by gerhard-boden** [comment]
> BTW: I consider the original argument of @gerhard-boden not that important nowadays. If you rely on the exact fusion pathes afx generates you are probably using it wrong anyways. Especially if anyone overrides fusion pathes generated by afx from the outside.

Whoa it's been a while since I composed that comment 😂  But I think it wasn't about overriding, but simply seeing what AFX argument (or path) caused the site to crash. Debugging named paths has always been easier for me.

Since the majority of more active AFX users than me seems to be okay with the dynamic approach it could easily be that I'm "using it wrong", so don't see my comment as a blocker in any way 😊

**Comment by mficzel** [comment]
Pretty sure the ide is applying .editorconfig rules to tests that verify that afx works with slightly messy code aswell.

I had to change the .editorconfig temporary to work on those tests.

**Comment by mficzel** [comment]
#

**Comment by mficzel** [comment]
Sorry for closing and opening, cellphones and GitHub do not mix well

File: Neos.Fusion.Afx/Classes/Service/AfxService.php (hunk 1 above)
Comment (mhsdesign) [reply Neos.Fusion.Afx/Classes/Service/AfxService.php]: done

File: Neos.Fusion.Afx/Tests/Functional/AfxServiceTest.php (hunk 2 above)
Comment (mhsdesign) [reply Neos.Fusion.Afx/Tests/Functional/AfxServiceTest.php]: thx done ;)

File: Neos.Fusion.Afx/Tests/Functional/AfxServiceTest.php (hunk 3 above)
Comment (mhsdesign) [reply Neos.Fusion.Afx/Tests/Functional/AfxServiceTest.php]: erledigt ^^

**Comment by mhsdesign** [comment]
okay all green now and stuff fixed.
Can we merge this soon before feature freeze and then we can (i will) address #3463

**Comment by markusguenther** [comment]
@jonnitto as the requested changes are outdated. Is this fine for you?

**Review by mficzel — APPROVED** [review]
I like this change, the code looks fine the teste make sense and it worked in my tests.

The readme could be adjusted but this may also happen later.

**Review by jonnitto — APPROVED** [review]

**Comment by jonnitto** [comment]
@mficzel Please add only issues to the project board… This makes it easier for us, to track the current state

**Pull request merged**